src/model/Course.java

```diff
@@ -12,36 +12,949 @@
+			SemesterOptions semester, int theoreticalHour, int labHour, CourseType type, Lecturer coordinator,
+			LinkedHashSet<Lecturer> lecturers, LinkedHashSet<Lecturer> assistants, String courseObjective,
+			LinkedHashSet<String> learningOutcomes, String courseDescription, CourseCategory category,
+			LinkedHashSet<WeeklySubject> schedule, CourseBook courseBook, LinkedHashSet<CourseBook> suggestedReading,
+			LinkedHashSet<EvaluationCriteria> evaluationCriterias, LinkedHashSet<SemesterActivity> workloadTable,
+			LinkedHashSet<CourseCompetency> courseCompetencies) {
+		this.courseName = courseName;
+		this.code = code;
+		this.creationDate = LocalDateTime.now();
+		this.prerequisites = new ArrayList<>(prerequisites);
+		this.courseLang = courseLang;
+		this.labHour = labHour;
+		this.theoreticalHour = theoreticalHour;
+		this.localCredit = calculateLocalCredit();
+		this.ects = calculateECTS();
+		this.type = type;
+		this.courseCordinator = coordinator;
+		this.courseLecturers = new ArrayList<>(lecturers);
+		this.assistants = new ArrayList<>(assistants);
+		this.courseObjective = courseObjective;
+		this.courseDescription = courseDescription;
+		this.learningOutcomes = new ArrayList<>(learningOutcomes);
+		this.courseCategory = category;
+		this.courseSchedule = new ArrayList<>(schedule);
+		this.courseTextBook = courseBook;
+		this.suggestedReading = new ArrayList<>(suggestedReading);
+		this.evaluationCriterias = new ArrayList<>(evaluationCriterias);
+		this.workloadTable = new ArrayList<>(workloadTable);
+		this.courseCompetencies = new ArrayList<>(courseCompetencies);
+	}
 	
+	//FIXME add a default constructor for newly created courses without a sufficient information supplied.
+
+	// Method(s)
+	/**
+	 * Add a prerequisite to the course.
+	 * 
+	 * @param courseCode Code of course that will be added as a prerequisites.
+	 * @return {@code true} if the new course added successfully to the
+	 *         {@link #prerequisites} list, return {@code false} otherwise.
+	 */
+	public boolean addPrerequisite(String courseCode) {
+		if (doesContain(prerequisites, courseCode)) {
+			return false;
+		} else {
+			return prerequisites.add(courseCode);
+		}
+
+	}
+
+	/**
+	 * This method removes a prerequisite from the {@link #prerequisites} list.
+	 * 
+	 * @param course
+	 * @return {@code true} if the deletion operation is successful, if the
+	 *         {@link #prerequisites} does not contain the specified course code it
+	 *         will return false.
+	 */
+	public boolean removePrerequisite(String course) {
+		return prerequisites.remove(course);
+	}
+
+	/**
+	 * It calculates ECTS after change in {@link #workloadTable}.
+	 * 
+	 * @return new ECTS value of the course after the change applied in the workload
+	 *         table.
+	 */
+	public int calculateECTS() {
+		int totalWorkload = 0;
+		for (SemesterActivity activity : workloadTable) {
+			totalWorkload += activity.getWorkload();
+		}
+		return ects = (int) Math.round(totalWorkload / 30.0);
```

Comment:
burada üstüne mi yuvarlıyorduk altına mı 🤔

Reply:
Kaya hoca aşağıya yuvarlamamızı istedi fakat aynı zamanda 148 iş yüküne sahip bir ders için 5 ECTS girmemizin daha doğru olacağını söyledi ilk projeyi anlattığı derslerde. Ben daha esnek bir yaklaşım ile yuvarlamayı tercih ettim fakat dediğini dikkate aldım. Bunu mail olarak iletip soralım eğer yuvarlamanın uygun olmayacağını düşünüyorsa değişiklik yapabiliriz.